tests/postgres_tests/test_aggregates.py

```diff
@@ -302,6 +302,23 @@ def test_array_agg_filter_slice(self):
         )
         self.assertCountEqual(qs, [[], [5]])
 
+    def test_array_agg_with_empty_filter_value(self):
+        with self.assertRaises(AssertionError) as context:
+            result = (
+                AggregateTestModel.objects.annotate(
+                    test_array_agg=ArrayAgg(
+                        "stattestmodel__int1",
+                        filter=Q(pk__in=[]),
+                        default=Value([]),
```

Comment:
An optional idea: maybe setup this as a subtest to test the conditions:

1. `[-1]`
2. `[]`
3. `Value([-1])`
4. `Value([])`

This increases our "black box" coverage.  (Simon mentioned that `Value(…)` is no longer necessary and we may want to update the docs to reflect that but it still may be valuable to use for coverage.)